micro-rdk/src/common/conn/viam.rs

```diff
@@ -489,14 +489,12 @@ where
         }
 
         // Since provisioning was run and completed, credentials are properly populated
-        // if wifi manager is configured loop forever until wifi is connected
+        // if wifi manager is configured loop forever until wifi is connected via
+        // a the provisioned network or one from previously stored agent config
         if let Some(wifi) = self.wifi_manager.as_ref().as_ref() {
-            while let Err(err) = wifi
-                .set_sta_mode(self.storage.get_wifi_credentials().unwrap())
-                .await
-            {
-                log::error!("couldn't connect to wifi reason {:?}", err);
-                let _ = Timer::after(Duration::from_secs(2)).await;
+            let networks = self.storage.get_all_networks().unwrap();
+            log::info!("attempting to configure wifi according to priority...");
+            while let Err(_) = wifi.try_connect_with_priority(networks.clone()).await {
```

Comment:
I'm not a fan of this `clone()`. We could remove it, however I'm not sure how we can ensure the inputs are sorted otherwise. same issue exists with `store_network_settings`.